enderio-machines/src/main/java/com/enderio/machines/common/blocks/base/blockentity/PoweredMachineBlockEntity.java

```diff
@@ -234,12 +234,14 @@ public final int getCapacitorSlotIndex() {
         return layout.getCapacitorSlot();
     }
 
-    private void updateCapacitorData() {
+    protected void updateCapacitorData() {
         isCapacitorDataDirty = false;
 
         if (supportsCapacitor()) {
             capacitorData = getCapacitorItem().getOrDefault(EIODataComponents.CAPACITOR_DATA, CapacitorData.NONE);
-
+            if (isCapacitorInstalled() && energyStorage.getEnergyStored() > energyStorage.getMaxEnergyStored()) {
```

Comment:
This isn't necessary - MachineEnergyStorage handles the stored amount exceeding the capacitor's maximum already.